### PR TITLE
feat: implement restriction to mobile devices

### LINKED ISSUE #388: [Frontend] Limit user to use mobile version from phone out of mini-app
🚨 Attention: Please read the guidelines before applying; otherwise, you will be ignored. 🚫📋
0. Please, join the telegram group to keep in touch https://t.me/+m0IGRiBvLtsyMzIy

1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Add popup window if user trying to use `web_site` from mobile phone.
3. Use this pop up window template with next text:
Title: Mobile website restriction 
Text: Please, use desktop version or telegram mini-app
Button left: cancel 
Button right: open in telegram (link to this bot `@spotnet_bot`)
Screenshot 2024-12-16 at 18 00 18

### PR DESCRIPTION
**Pull request by Akshola00**
resolves #388

### CODE REVIEW

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -69,6 +72,35 @@ function App() {
...
+      const isMobileWidth = window.innerWidth <= 768;
+      setIsMobile(isMobileDevice || isMobileWidth);
+    };
+
+    checkMobile();
+
+    window.addEventListener('resize', checkMobile);
+
+    return () => window.removeEventListener('resize', checkMobile);
+  }, []);
```
Comment (whateverfw) [inline frontend/src/App.jsx:94]: this should be a custom hook

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -109,6 +141,20 @@ function App() {
         </Routes>
       </main>
       <Footer />
+      {isMobile && (
+         <ActionModal
+         isOpen={modal}
+         title="Mobile website restriction"
+         subTitle=""
+         content={[
```
Comment (whateverfw) [inline frontend/src/App.jsx:149]: please, put the text into sub-title. not content

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -69,6 +72,35 @@ function App() {
...
+
+    return () => window.removeEventListener('resize', checkMobile);
+  }, []);
+
+  const handleCloseModal = () => {
+    setModal(false);
+  };
+
+  const openTelegramBot = () => {
+    const telegramBotLink = 'https://t.me/spotnet_bot';
```
Comment (whateverfw) [inline frontend/src/App.jsx:101]: extract it into `frontend/src/utils/constants.js`, rename to be in constant style (uppercase) as well

File: frontend/src/App.jsx [hunk 4]
```diff
@@ -24,6 +24,9 @@ function App() {
   const [tgUser, setTgUser] = useState(JSON.parse(localStorage.getItem('tg_user')));
   const [showModal, setShowModal] = useState(false);
   const navigate = useNavigate();
+  const [isMobile, setIsMobile] = useState(false);
+  const [modal, setModal] = useState(true);
```
Comment (whateverfw) [inline frontend/src/App.jsx:28]: rename it to be something more meaningful, like [isMobileRestrictionModalOpen, ..]

**Review by whateverfw — CHANGES_REQUESTED** [review]

File: frontend/src/App.jsx (hunk 4 above)
Comment (Akshola00) [reply frontend/src/App.jsx:28]: renamed

File: frontend/src/App.jsx (hunk 2 above)
Comment (Akshola00) [reply frontend/src/App.jsx:149]: done

File: frontend/package-lock.json
Comment (whateverfw) [inline frontend/package-lock.json:1]: remove this file, use yarn please

**Review by whateverfw — CHANGES_REQUESTED** [review]
Please remove package-lock.json and resolve merge conflict. Apart from that everything is fine

**Comment by Akshola00** [comment]
@whateverfw changes made

**Review by whateverfw — APPROVED** [review]
@Akshola00 @djeck1432 lgtm

**Review by djeck1432 — CHANGES_REQUESTED** [review]
Can you please add env variable, like
DISABLE_DESKTOP_ON_MOBILE and if it's True, then we will hide it, if false we will shot it. Just in case we will need to have it under control

**Comment by whateverfw** [comment]
@Akshola00 ⏫

**Comment by Akshola00** [comment]
Changes made

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**